lib/router/transition.ts

```diff
@@ -239,19 +233,30 @@ export default class Transition<T extends Route> implements Partial<Promise<T>>
     @public
    */
   abort() {
-    if (this.isAborted) {
-      return this;
-    }
-    log(this.router, this.sequence, this.targetName + ': transition was aborted');
-    if (this.intent !== undefined && this.intent !== null) {
-      this.intent.preTransitionState = this.router.state;
-    }
-    this.isAborted = true;
-    this.isActive = false;
-    this.router.activeTransition = undefined;
+    this.rollback();
+    this.to = this.from;
```

Comment:
This seems slightly odd to me. Don't we want to essentially model `Transition`'s as immutable (at least their from/to info) and create _new_ transition instances when we want to mutate?